src/components/Schedule/TableView.tsx

```diff
@@ -1,15 +1,172 @@
+  const appointments = courses.flatMap((course) => {
+    const professors = course.professors.map((prof) => prof.username); //TO-DO switch to displayName, when its in schema
+
+    const assignedCourses = splitCourseDays(course).map((meetingTime) => {
+      return {
+        courseTitle: course.CourseID.code, //not in schema yet
+        courseNumber: course.CourseID.code,
+        subject: course.CourseID.subject,
+        section: "TEMP_SECTION", //not in schema yet
+        prof: professors,
+        classSize: course.capacity,
+        startDate: meetingTime.startDate,
+        endDate: meetingTime.endDate,
+      } as Appointment;
+    });
+
+    return assignedCourses;
+  });
+  return appointments;
+};
+
+const populateTable = (courses: Appointment[]) => {
+  var table_b = document.getElementById("table_body");
+  if (table_b != null) {
+    //remove previous table
+    if (table_b.childNodes.length !== 0) {
+      while (table_b.childNodes.length > 0) {
+        table_b.childNodes[0].remove();
+      }
+    }
+    var nTd, nTr;
+    for (var i = 0; i < courses.length; i++) {
+      nTr = document.createElement("Tr");
+
+      var sdate = courses[i].startDate
+        .toString()
+        .split(" ")
+        .slice(0, 4)
+        .join(" ");
+      var stime = courses[i].startDate
+        .toString()
+        .split(" ")
+        .slice(4, 5)
+        .join(" ");
+      var edate = courses[i].endDate
+        .toString()
+        .split(" ")
+        .slice(0, 4)
+        .join(" ");
+      var etime = courses[i].endDate
+        .toString()
+        .split(" ")
+        .slice(4, 5)
+        .join(" ");
+
+      //Course title
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = courses[i].subject + " " + courses[i].courseTitle;
+      nTr.appendChild(nTd);
+
+      //Schedule time
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = stime + " / " + etime;
+      nTr.appendChild(nTd);
+
+      //Term
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = "Summer";
+      nTr.appendChild(nTd);
+
+      //Prof
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = courses[i].prof[0];
+      nTr.appendChild(nTd);
+
+      //Course Code
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = courses[i].section;
+      nTr.appendChild(nTd);
+
+      //Start/End Data
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = sdate + " / " + edate;
+      nTr.appendChild(nTd);
+
+      //# of students
+      nTd = document.createElement("Td");
+      nTd.style.padding = "8px 16px";
+      nTd.innerText = courses[i].classSize.toString();
+      nTr.appendChild(nTd);
+      table_b?.appendChild(nTr);
+    }
```

Comment:
You can replace all of this with just a `courses.map` to convert all of these into React components. We definitely want to be using Chakra components for everything as that way we can manage the themeing and consistency of everything throughout the application

Reply:
Second this, you can see some examples that I used map for in SurveyResults and ProfileManagement

Reply:
Sounds good I will work on this! thanks for the suggestion

Reply:
@Adamk8 I am having trouble dynamically mapping each new row of the table. My understanding is I should be adding entries into the tableData and then it should automatically map it to the table, is that correct or am I missing something? I also tried dynamically creating chakra elements but it does not allow appendingchilds so im a bit stuck